sceptre/cli/template.py

```diff
@@ -100,3 +100,65 @@ def estimate_cost_command(ctx, path):
             response = response["Url"]
             webbrowser.open(response, new=2)
         write(response + "\n", 'text')
+
+
+@click.command(name="fetch-remote-template", short_help="Prints the remote template.")
+@click.argument("path")
+@click.pass_context
+@catch_exceptions
+def fetch_remote_template_command(ctx, path):
+    """
+    Prints the remote template used for stack in PATH.
+    \f
+
+    :param path: Path to execute the command on.
+    :type path: str
+    """
+    context = SceptreContext(
+        command_path=path,
+        project_path=ctx.obj.get("project_path"),
+        user_variables=ctx.obj.get("user_variables"),
+        options=ctx.obj.get("options"),
+        output_format=ctx.obj.get("output_format"),
+        ignore_dependencies=ctx.obj.get("ignore_dependencies")
+    )
+
+    plan = SceptrePlan(context)
+    responses = plan.fetch_remote_template()
+    output = [template for template in responses.values()]
+    write(output, context.output_format)
+
+
+@click.command(name="diff", short_help="Show diffs with running stack.")
+@click.argument("path")
+@click.option(
+    "-D", "--differ", type=click.Choice(["difflib", "dictdiffer"]),
+    default="difflib", help="Specify diff library, default difflib."
+)
+@click.pass_context
+@catch_exceptions
+def diff_command(ctx, path, differ):
+    """
+    Show diffs between the running and generated stack.
+    \f
+
+    :param path: The path to execute the command on.
+    :type path: str
+    :param differ: The diff library to use, default difflib.
+    :type differ: str
+    """
+    context = SceptreContext(
+        command_path=path,
+        project_path=ctx.obj.get("project_path"),
+        user_variables=ctx.obj.get("user_variables"),
+        options=ctx.obj.get("options"),
+        ignore_dependencies=ctx.obj.get("ignore_dependencies")
+    )
+
+    plan = SceptrePlan(context)
+    responses = plan.diff(differ)
+    output = "\n".join([
+        stack_name + ": " + template
+        for stack_name, template in responses.values()
+    ])
+    write(output, context.output_format)
```

Comment:
I think there's a real need to be able to output the diff to a file. If you rely only on stdout, that will include all sorts of logs and printed statements from sceptre and hooks and such... it doesn't make for a "clean" output that can be posted somewhere for review by the likes of Jenkins or whatever

Reply:
With this implementation, so long as `--debug` isn't specified, only the stack name and then the diff is written to stdout.

Reply:
So we have hooks that do things like SAM Build / Package and printing outputs. What could be done is it defaults to stdout, but you can pass an `-o/--output-file` argument that instead writes to that file. This way we can output that file easily enough.